src/dbnode/storage/index.go

```diff
@@ -1495,12 +1510,12 @@ func (i *nsIndex) overriddenOptsForQueryWithRLock(
 	opts index.QueryOptions,
 ) index.QueryOptions {
 	// Override query response limit if needed.
-	if i.state.runtimeOpts.maxQueryLimit > 0 && (opts.Limit == 0 ||
-		int64(opts.Limit) > i.state.runtimeOpts.maxQueryLimit) {
+	if i.state.runtimeOpts.maxQueryLimit > 0 && (opts.SeriesLimit == 0 ||
```

Comment:
Should we also override the doc limit? We use this override sometimes for investigating problems, but may be difficult to do if we can't tweak the document limit